src/common/host_device_vector.cu

```diff
@@ -412,6 +413,7 @@ struct HostDeviceVectorImpl {
       perm_h_.Grant(access);
       return;
     }
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
I don't see how this line is relevant to the CUDF support. Is this line response to a known concurrency bug?

Reply:
@hcho3 This is #4751 .  Before we don't have data source in GPU, after this PR we have and it's blocking me.